lib/manageiq/rpm_build/generate_core.rb

```diff
@@ -97,6 +99,34 @@ def plugin_paths
         end
       end
 
+      def fixup_sti_loader!
+        sti_loader_yml_path = miq_dir.join("tmp/cache/sti_loader.yml")
+
+        sti_loader = YAML.load_file(sti_loader_yml_path)
+        sti_loader.transform_keys! do |path|
+          if !path.start_with?(BUILD_DIR.to_s)
+            path
+          else
+            rel_path = Pathname.new(path).relative_path_from(BUILD_DIR)
+            prefix, target_path = rel_path.to_s.split("/", 2)
+
+            prefix =
+              case prefix
+              when OPTIONS.product_name
+                "/var/www/miq/vmdb"
+              when /^#{OPTIONS.product_name}-gemset/
+                File.join("", "opt", OPTIONS.rpm.org_name, "#{OPTIONS.product_name}-gemset")
+              else
+                raise "Invalid file path in STI cache: #{path}"
+              end
+
+            File.join(prefix, target_path)
+          end
+        end
+
+        File.write(sti_loader_yml_path, sti_loader.to_yaml)
```

Comment:
NOTE I need to re-set the file's mtime

Reply:
Do you?  I don't think the mtime of the sti_loader.yml file itself matters.

Reply:
as long as the files have not been modified between the time this was originally written and the new time it is written, I think this is probably fine.

Reply:
yeah you're right I'm still stuck on the mtime of the sti_loader.yml